frontend/src/App.vue

```diff
@@ -54,16 +53,67 @@ export default {
       );
     },
   },
-  watch: {
-  },
+  watch: {},
   methods: {
     async onAction(contextualItem, index) {
       const itemFqid = contextualItem?.item?.fqid;
-      await this.rpc.invoke("performAction", [itemFqid, index, contextualItem.context]);
+      await this.rpc.invoke("performAction", [
+        itemFqid,
+        index,
+        contextualItem.context,
+      ]);
     },
+  getItem(itemFqid) {
+    for (const collection of this.collections) {
```

Comment:
The deep search logic seems overly complicated.
1. handling at the root (collection.items) level using `for of` loop.
2. handling 1 layer deeper (2nd layer) using `for of` loop.
3. from 3rd layer invoke a DFS recursive algo

Reply:
My naive approach to performing a **deep** nested search would be:

To first collect **all** the items in the entire tree by aggregating
from the **current** "layer" of the tree, while "recording" the **next** layer  for the tree
that would be explored in the next iteration.

Secondly, now that we have all the items in the tree, we can perform a simple `.find()`
on this `flattened` array.

Basically doing a BFS + flattening results + find().